extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1289,8 +1295,8 @@ private long getOffsetFromStorageForPartition(int partition)
       long latestKafkaOffset = getOffsetFromKafkaForPartition(partition, false);
       if (offset > latestKafkaOffset) {
         throw new ISE(
-            "Offset in metadata storage [%,d] > latest Kafka offset [%,d] for partition [%d]. If your Kafka offsets have"
-            + " been reset, you will need to remove the entry for [%s] from the dataSource table.",
+            "Offset in metadata storage [%,d] > latest Kafka offset [%,d] for partition[%d] dataSource[%s]. If your "
+            + "Kafka offsets have been reset, you will need to use the supervisor reset API to remove the old entry.",
```

Comment:
Potential alternate wording:

"If these messages are no longer available (perhaps you deleted and re-created your Kafka topic) you can use the supervisor reset API to restart ingestion."

your choice if you want to use it or not.